alpha/lib/model/entry.php

```diff
@@ -3342,4 +3359,62 @@ public function getUserNames() {
 		
 		return implode(" ", $userNames);
 	}
+
+	public function getcapabilities()
+	{
+
+		$capabilitiesUnserialized = $this->getFromCustomData(self::CAPABILITIES);
+		if (is_null($capabilitiesUnserialized))
+		{
+			return "";
+		}
+		else
+		{
+			$capabilitiesArr = unserialize($capabilitiesUnserialized);
+		}
+		$capabilitiesStr = "";
+		$first = true;
+		foreach ($capabilitiesArr as $capability)
+		{
+			if ($first)
+			{
+				$first = false;
+			} else
+			{
+				$capabilitiesStr .= ",";
+			}
+			$capabilitiesStr .= $capability;
+		}
+		return $capabilitiesStr;
+	}
+
+	public function addCapability($capability)
+	{
+		$capabilitiesStr = $this->getFromCustomData(self::CAPABILITIES);
+		if (is_null($capabilitiesStr))
+		{
+			$capabilitiesArr = array();
+		}
+		else
+		{
+			$capabilitiesArr = unserialize($capabilitiesStr);
+		}
+		$arrSpot = array_search($capability, array_values($capabilitiesArr));
```

Comment:
Implement with keys, it will save you the array_search, also use array() as default value of getFromCustomData, it will save you the is_null.